modules/services/views/src/main/java/au/com/shiftyjelly/pocketcasts/views/multiselect/MultiSelectAction.kt

```diff
@@ -108,59 +26,185 @@ sealed class MultiSelectAction(
         "select_all"
     )
 
+    sealed class BaseEpisodeAction(
+        override val groupId: Int,
+        override val actionId: Int,
+        @StringRes override val title: Int,
+        @DrawableRes override val iconRes: Int,
+        override val analyticsValue: String,
+        override val isVisible: Boolean = true,
+    ) : MultiSelectAction(
+        groupId,
+        actionId,
+        title,
+        iconRes,
+        analyticsValue,
+        isVisible
+    ) {
+        object DeleteDownload : BaseEpisodeAction(
+            R.id.menu_download,
+            UR.id.menu_undownload,
+            LR.string.delete_download,
+            IR.drawable.ic_undownload,
+            "remove_download"
+        )
+        object Download : BaseEpisodeAction(
+            R.id.menu_download,
+            R.id.menu_download,
+            LR.string.download,
+            IR.drawable.ic_download,
+            "download"
+        )
+        object Archive : BaseEpisodeAction(
+            R.id.menu_archive,
+            R.id.menu_archive,
+            LR.string.archive,
+            IR.drawable.ic_archive,
+            "archive"
+        )
+        object Unarchive : BaseEpisodeAction(
+            R.id.menu_archive,
+            UR.id.menu_unarchive,
+            LR.string.unarchive,
+            IR.drawable.ic_unarchive,
+            "unarchive"
+        )
+        object DeleteUserEpisode : BaseEpisodeAction(
+            R.id.menu_archive,
+            R.id.menu_delete,
+            LR.string.delete,
+            R.drawable.ic_delete,
+            "delete"
+        )
+        object MarkAsUnplayed : BaseEpisodeAction(
+            R.id.menu_mark_played,
+            UR.id.menu_markasunplayed,
+            LR.string.mark_as_unplayed,
+            IR.drawable.ic_markasunplayed,
+            "mark_as_unplayed"
+        )
+        object MarkAsPlayed : BaseEpisodeAction(
+            R.id.menu_mark_played,
+            R.id.menu_mark_played,
+            LR.string.mark_as_played,
+            IR.drawable.ic_markasplayed,
+            "mark_as_played"
+        )
+        object PlayNext : BaseEpisodeAction(
+            R.id.menu_playnext,
+            R.id.menu_playnext,
+            LR.string.play_next,
+            IR.drawable.ic_upnext_playnext,
+            "play_next"
+        )
+        object PlayLast : BaseEpisodeAction(
+            R.id.menu_playlast,
+            R.id.menu_playlast,
+            LR.string.play_last,
+            IR.drawable.ic_upnext_playlast,
+            "play_last"
+        )
+        object Unstar : BaseEpisodeAction(
+            R.id.menu_star,
+            UR.id.menu_unstar,
+            LR.string.unstar,
+            IR.drawable.ic_unstar,
+            "unstar"
+        )
+        object Star : BaseEpisodeAction(
+            R.id.menu_star,
+            R.id.menu_star,
+            LR.string.star,
+            IR.drawable.ic_star,
+            "star"
+        )
+    }
+
+    sealed class BookmarkAction(
```

Comment:
Grouped episode and bookmark actions inside `BaseEpisodeAction` and `BookmarkAction` respectively.